defaults/main.yml

```diff
@@ -116,6 +116,16 @@ checkmk_agent__user_home: '/var/lib/nagios'
 # SSH Public key for logging in with SSH user.
 checkmk_agent__user_key: ''
 
+
+# .. envvar:: checkmk_agent__authorized_keys__dependent_list
+#
+# Authorized key configuration for the ``debops.authorized_keys`` role.
+checkmk_agent__authorized_keys__dependent_list:
+  - name: '{{ checkmk_agent__ssh_user }}'
+    ssh_keys: [ '{{ checkmk_agent__user_key }} if {{ checkmk_agent__user_key|d() }} else {{ hostvars[groups["debops_service_checkmk_server"][0]].ansible_local.checkmk_server.ssh_publickey|d("") }}' ]
```

Comment:
What deprecated warning, about Undefined variable? This will be error in the future, you need to fix that.

Moving the lookup code to the variable is good, then users can modify it if they wish without the need to modify the role itself.

As for the 'role/env' solution - this comes from a pretty specific design principles of DebOps roles - they are supposed to be read-only, and you should pass configuration of other roles through default variables. In specific situations, like generating data on a host by your role, or using lookup templates, you might find that Ansible does not have the data it needs at this specific point in the playbook, or it does the lookup in the wrong role.

In the given `debops.php` example, the role checks what PHP package versions are available and then acts on that information, however another role, `debops.logrotate` is also using this information to generate the correct `logrotate` configuration for the `php-fpm` service. The information about PHP version is stored in a variable, however only `debops.php` sets this variable. Since `debops.logrotate` is supposed to run before `debops.php`, it would not know beforehand what PHP version it should use to generate the config file. Basically a casuality violation. :-)

To resolve this, I decided to split the `debops.php` role essentially into two - first one, `debops.php/env` contains minimum amount of tasks to "set the environment" for other roles - it checks what PHP versions are available, and even configures external APT repositories. When its tasks are done, other roles like `debops.logrotate` can act on variables passed through `debops.php` defaults since the required facts are set in the memory. After that, `debops.php` continues as usual to finish the configuration.

As I said, this is a very situational solution and you won't need this with the most roles. However, I suggested it here since you mentioned that the SSH keys on the specific hosts are generated by the role. To avioid the need to run the playbook twice to complete the configuration, you can do it like this:

- `checkmk_server/env` generates the SSH keys and slurps them into memory
- `debops.authorized_keys` configures the SSH keys using data from `checkmk_server` defaults + SSH keys
- `checkmk_server` finishes the configuration

This should keep things clean and consistent.